action_server/src/action_server/client.py

```diff
@@ -139,6 +139,10 @@ def send_task(self, semantics):
             self.cancel_all()
             raise KeyboardInterrupt
 
+        if not isinstance(result, action_server_msgs.msg.TaskResult):
+            rospy.logerr("Result not instance of 'action_server_msgs.msg.TaskResult', but {}".format(type(result)))
+            return False
+
```

Comment:
You only need to catch the result is None, right. Other types cannot (should not) be sent over the result.

Reply:
True, if everything is correct, the result is either a `TaskResult` message or `None`. But it is also true that in any other case than a `TaskResult` message, we should not continue to `task_outcome_from_result`.
So it is better to catch all other types.

Reply:
But we still want to make a distinction between None, which can happen and something else, which should not happen and means a bug. This should be reflected in the log.
So what I would suggest is in case of None don't log an error.

Reply:
Done.